Internal transaction indexer task is not reducing the `block_process_limit` with unexpected errors

**Describe the bug**
`index_internal_txs_task` is not changing the block process limit when an unexpected error happens. 
```

2023-05-30 15:26:07,460 [ERROR] [ddcbde5f/index_internal_txs_task] Task safe_transaction_service.history.tasks.index_internal_txs_task[ddcbde5f-3948-49fa-aea3-3d448bd8e915] raised unexpected: ValueError({'code': -32005, 'data': {'from': '0x629067', 'limit': 10000, 'to': '0x63C494'}, 'message': 'query returned more than 10000 results. Try with this block range [0x629067, 0x63C494].'})
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/celery/app/trace.py", line 451, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/celery/app/trace.py", line 734, in __protected_call__
    return self.run(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/celery/app/autoretry.py", line 34, in run
    return task._orig_run(*args, **kwargs)
  File "/app/safe_transaction_service/history/tasks.py", line 190, in index_internal_txs_task
    ) = InternalTxIndexerProvider().start()
  File "/app/safe_transaction_service/history/indexers/ethereum_indexer.py", line 498, in start
    ) = self.process_addresses(
  File "/app/safe_transaction_service/history/indexers/ethereum_indexer.py", line 390, in process_addresses
    elements = self.find_relevant_elements(
  File "/app/safe_transaction_service/history/indexers/internal_tx_indexer.py", line 118, in find_relevant_elements
    return self._find_relevant_elements_using_trace_filter(
  File "/app/safe_transaction_service/history/indexers/internal_tx_indexer.py", line 203, in _find_relevant_elements_using_trace_filter
    to_traces = self.ethereum_client.tracing.trace_filter(
  File "/usr/local/lib/python3.10/site-packages/gnosis/eth/ethereum_client.py", line 1145, in trace_filter
    return self.slow_w3.tracing.trace_filter(parameters)
  File "/usr/local/lib/python3.10/site-packages/web3/module.py", line 68, in caller
    result = w3.manager.request_blocking(
  File "/usr/local/lib/python3.10/site-packages/web3/manager.py", line 232, in request_blocking
    return self.formatted_response(
  File "/usr/local/lib/python3.10/site-packages/web3/manager.py", line 205, in formatted_response
    raise ValueError(response["error"])
ValueError: {'code': -32005, 'data': {'from': '0x629067', 'limit': 10000, 'to': '0x63C494'}, 'message': 'query returned more than 10000 results. Try with this block range [0x629067, 0x63C494].'}
```

**Expected behavior**
Catch generic exception and force the block_process_limit to 1. 

**Aditional information** 
erc20 indexer catch the error
```
index_erc20_events_task] Erc20EventsIndexer: Value error retrieving events from-block=9091977 to-block=9091987 : {'code': -32005, 'data': {'from': '0x8ABB89', 'limit': 10000, 'to': '0x8ABB92'}, 'message': 'query returned more than 10000 results. Try with this block range [0x8ABB89, 0x8ABB92].'}
2023-05-30 15:50:09,816 [INFO] [06a9d862/index_erc20_events_task] Erc20EventsIndexer: block_process_limit set back to 1
```

--- a/safe_transaction_service/history/indexers/internal_tx_indexer.py
+++ b/safe_transaction_service/history/indexers/internal_tx_indexer.py
@@ -209,6 +209,20 @@ class InternalTxIndexer(EthereumIndexer):
             raise FindRelevantElementsException(
                 "Request error calling `trace_filter`"
             ) from e
+        except ValueError as e:
+            # For example, Goerli returns:
+            #   ValueError: {'code': -32005, 'data': {'from': '0x6BBCE1', 'limit': 10000, 'to': '0x7072DB'}, 'message': 'query returned more than 10000 results. Try with this block range [0x6BBCE1, 0x7072DB].'}
+            logger.warning(
+                "%s: Value error retrieving trace_filter results from-block=%d to-block=%d : %s",
+                self.__class__.__name__,
+                from_block_number,
+                to_block_number,
+                e,
+            )
+            raise FindRelevantElementsException(
+                f"Request error retrieving trace_filter results "
+                f"from-block={from_block_number} to-block={to_block_number}"
+            ) from e
 
         # Log INFO if traces found, DEBUG if not
         traces: OrderedDict[HexBytes, None] = OrderedDict()

--- a/safe_transaction_service/history/indexers/internal_tx_indexer.py
+++ b/safe_transaction_service/history/indexers/internal_tx_indexer.py
@@ -210,7 +210,7 @@ class InternalTxIndexer(EthereumIndexer):
                 "Request error calling `trace_filter`"
             ) from e
         except ValueError as e:
-            # For example, Goerli returns:
+            # For example, Infura returns:
             #   ValueError: {'code': -32005, 'data': {'from': '0x6BBCE1', 'limit': 10000, 'to': '0x7072DB'}, 'message': 'query returned more than 10000 results. Try with this block range [0x6BBCE1, 0x7072DB].'}
             logger.warning(
                 "%s: Value error retrieving trace_filter results from-block=%d to-block=%d : %s",

